### PR TITLE
Add: Full mouse support for rnvimr

### PR DESCRIPTION
**Pull request by Zeioth**
I hope this makes easier for you to test the feature. Unless you wanna add an option to enable/disable mouse support (which shouldn't really be necessary), that should be all.

### CODE REVIEW

File: autoload/rnvimr.vim [hunk 1]
```diff
@@ -213,6 +241,7 @@ function! rnvimr#init(...) abort
     if !empty(urc_path)
         let env.RNVIMR_URC_PATH = urc_path
     endif
+    rnvimr#enable_mouse_support()
```
Comment (kevinhwang91) [inline autoload/rnvimr.vim:245]: missing `call`

File: autoload/rnvimr.vim [hunk 2]
```diff
@@ -182,6 +182,34 @@ function! rnvimr#open(path) abort
...
+        augroup end
+
+        if match(n_mouse, '[a|h|n]') >= 0
+            augroup RnvimrMouse
+                autocmd TermEnter,WinEnter <buffer> call nvim_set_option('mouse', '')
+                execute printf("autocmd WinLeave <buffer> call nvim_set_option('mouse', '%s')", n_mouse)
+            augroup END
+        endif
+
+        if system('tmux display -p "#{mouse}"')[0]
```
Comment (kevinhwang91) [inline autoload/rnvimr.vim:200]: Should use `TMUX` var
> The update-environment session option may be used to update the session environment from the client when a new session is created or an old reattached.
> tmux also initialises the TMUX variable with some internal information to allow commands to be executed from inside, and the TERM variable with the correct terminal setting of ‘screen’.

**Review by kevinhwang91 — CHANGES_REQUESTED** [review]

File: autoload/rnvimr.vim (hunk 1 above)
Comment (Zeioth) [reply autoload/rnvimr.vim:247]: Change applied.

File: autoload/rnvimr.vim (hunk 2 above)
Comment (Zeioth) [reply autoload/rnvimr.vim]: Change applyed

**Comment by kevinhwang91** [comment]
Thanks!

**Pull request merged**